pkg/internal/cyberark/servicediscovery/discovery.go

```diff
@@ -0,0 +1,132 @@
+package servicediscovery
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/jetstack/preflight/pkg/version"
+)
+
+const (
+	prodDiscoveryEndpoint        = "https://platform-discovery.cyberark.cloud/api/v2/"
+	integrationDiscoveryEndpoint = "https://platform-discovery.integration-cyberark.cloud/api/v2/"
+
+	// identityServiceName is the name of the identity service we're looking for in responses from the Service Discovery API
+	// TODO: confirm that this is definitely the one to use
```

Comment:
nb: this is a TODO, but I don't think it blocks merging this. We'll confirm ~next week, and update then.

Reply:
Had the correct one confirmed, so this is now done. I've updated the PR